app/src/org/commcare/appupdate/AppUpdateControllerFactory.java

```diff
@@ -0,0 +1,21 @@
+package org.commcare.appupdate;
+
+import android.content.Context;
+import com.google.android.play.core.appupdate.AppUpdateManagerFactory;
+
+/**
+ * A factory that creates {@link AppUpdateController} instance.
+ * @author $|-|!˅@M
+ */
+public class AppUpdateControllerFactory {
+
+    /**
+     * @param callback {@link Runnable} to notify when there is a change in update state.
+     * @param context Application context of app.
+     * @return A new {@link FlexibleAppUpdateController} to use for in-app updates
+     */
+    public static FlexibleAppUpdateController create(Runnable callback, Context context) {
+        // TODO: Should we check in-app update is enabled and return a demo or full operational controller?
+        return new CommcareFlexibleAppUpdateManager(callback, AppUpdateManagerFactory.create(context));
```

Comment:
Only using flexbile update type here cuz as per [this issue](https://issuetracker.google.com/issues/133299031) it's upto developer to choose the type of update.